src/expr/impl/src/scalar/array.rs

```diff
@@ -28,6 +29,41 @@ fn row_(row: impl Row) -> StructValue {
     StructValue::new(row.iter().map(|d| d.to_owned_datum()).collect())
 }
 
+/// # Example
+///
+/// ```slt
+/// query T
+/// select map_from_entries(null::int[], array[1,2,3]);
+/// ----
+/// NULL
+///
+/// query T
+/// select map_from_entries(array['a','b','c'], array[1,2,3]);
+/// ----
+/// {"a":1,"b":2,"c":3}
+/// ```
+#[function(
+    "map_from_entries(anyarray, anyarray) -> anymap",
+    type_infer = "|args| Ok(MapType::from_kv(args[0].as_list().clone(), args[1].as_list().clone()).into())"
+)]
+fn map(key: ListRef<'_>, value: ListRef<'_>) -> Result<MapValue, ExprError> {
```

Comment:
I suppose we can theoretically return a `MapRef` here with zero-copy? 🤔

Reply:
discuss on the comment of MapRef